docs/async/async-in-depth.md

```diff
@@ -0,0 +1,151 @@
+
+Using `await` allows your application or service to perform useful work while a task is running by yielding control to its caller until the task is done. Your code does not need to rely on callbacks or events to continue execution after the task has been completed. The language and task API integration does that for you. If you’re using `Task<T>`, the `await` keyword will additionally “unwrap” the value returned when the Task is complete.  The details of how this works are explained further below.
+
+You can learn more about tasks and the different ways to interact with them in the [Task-based Asynchronous Pattern (TAP) Article](https://msdn.microsoft.com/en-us/library/hh873175(v=vs.110).aspx).
+
+## Deeper Dive into Tasks for an I/O-Bound Operation
+
+The following section describes a 10,000 foot view of what happens with a typical async I/O call. Let's start with a couple examples.
+
+The first example calls an async method and returns an active task, likely yet to complete.
+
+```csharp
+public Task<string> GetHtmlAsync()
+{
+ 	// Execution is synchronous here
+	var client = new HttpClient();
+	
+	return client.GetStringAsync("http://www.dotnetfoundation.org");
+}
+```
+
+The second example adds the use of he `async` and `await` keywords to operate on the task.
+
+```csharp
+public async Task<string> GetFirstCharactersCountAsync(string url, int count)
+{
+	// Execution is synchronous here
+	var client = new HttpClient();
+	
+	// Execution of GetFirstCharactersCountAsync() is yielded to the caller here
+	// GetStringAsync returns a Task<string>, which is *awaited*
+	var page = await client.GetStringAsync("http://www.dotnetfoundation.org");
+	
+	// Execution resumes when the client.GetStringAsync task completes,
+    // becoming synchronous again.
+	
+	if (count > page.Length)
+	{
+		return page;
+	}
+	else
+	{
+		return page.Substring(0, count);
+	}
+}
+```
+
+The call to `GetStringAsync()` calls through lower-level .NET libraries (perhaps calling other async methods) until it reaches a P/Invoke interop call into a native networking library. The native library may subsequently call into a System API call (such as `write()` to a socket on Linux). A task object will be created at the native/managed bounary, possibly using [TaskCompletionSource](https://msdn.microsoft.com/en-us/library/dd449202(v=vs.110).aspx). The task object will be passed up through the layers, possibly operated on or directly returned, eventually returned to the initial caller. 
+
+In the second example above, a `Task<T>` object will be returned from `GetStringAsync`. The use of the `await` keyword will result in control being yielded to the caller as well as a newly created task object, required to monitor progress of the `GetFirstCharactersCountAsync` method. The `await` keyword also unwraps the return value from the task object, from [Task.Result](https://msdn.microsoft.com/en-us/library/dd321468(v=vs.110).aspx).
```

Comment:
The second sentence reads very awkwardly to me. I'd suggest:
'The use of the await keyword causes the method to return a newly created task object. Control returns to the caller from this location in the GetFirstCharactersCountAsync method. The methods and properties of the Task object enable callers to monitor the progress of the task, which will complete when the remaining code in GetFirstCharactersCountAsync has executed.